packages/core/src/error.ts

```diff
@@ -90,8 +109,62 @@ function newError (message: string, code?: Neo4jErrorCode): Neo4jError {
   return new Neo4jError(message, code ?? NOT_AVAILABLE)
 }
 
+/**
+ * Verifies if the given error is retriable.
+ *
+ * @public
+ * @param {object|undefined|null} error the error object
+ * @returns {boolean} true if the error is retriable
+ */
+ const isRetriableError = Neo4jError.isRetriable
+
+/**
+ * @private
+ * @param {string} code the error code
+ * @returns {boolean} true if the error is a retriable error
+ */
+function _isRetriableCode (code?: Neo4jErrorCode): boolean {
+  return code === SERVICE_UNAVAILABLE ||
+    code === SESSION_EXPIRED ||
+    _isAuthorizationExpired(code) ||
+    _isTransientError(code)
+}
+
+/**
+ * @private
+ * @param {string} code the error to check
+ * @return {boolean} true if the error is a transient error
+ */
+function _isTransientError (code?: Neo4jErrorCode): boolean {
```

Comment:
Please note that we are currently looking at whether we might be able to change these 2 errors to a different category.

Reply:
Sure, this function is internal, but I will change the name to something like `_isRetriableTransientError()`.